owncloudData/src/androidTest/java/com/owncloud/android/data/roommigrations/MigrationToDB28.kt

```diff
@@ -37,159 +33,56 @@ import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.CAPABILITIES_VER
 import com.owncloud.android.data.capabilities.db.OCCapabilityEntity.Companion.toContentValues
-import com.owncloud.android.domain.capabilities.model.CapabilityBooleanType.Companion.capabilityBooleanTypeUnknownInt
 import com.owncloud.android.testutil.OC_CAPABILITY
 import org.junit.Assert.assertEquals
-import org.junit.Rule
 import org.junit.Test
-import java.io.IOException
 
+/**
+ * Test the migration from database to version 28.
+ */
 @SmallTest
-class MigrationToDB28 {
-
-    @Rule
-    @JvmField
-    val helper: MigrationTestHelper = MigrationTestHelper(
-        InstrumentationRegistry.getInstrumentation(),
-        OwncloudDatabase::class.java.canonicalName,
-        FrameworkSQLiteOpenHelperFactory()
-    )
+class MigrationToDB28 : MigrationTest() {
 
     @Test
-    @Throws(IOException::class)
     fun migrate27To28() {
-        with(
-            helper.createDatabase(
-                TEST_DB_NAME,
-                DB_VERSION_27
-            )
-        ) {
-            insert(CAPABILITIES_TABLE_NAME, SQLiteDatabase.CONFLICT_NONE, cv)
-            insert(CAPABILITIES_TABLE_NAME, SQLiteDatabase.CONFLICT_NONE, cvWithDefaultValues)
-            close()
+        helper.createDatabase(
+            TEST_DB_NAME,
+            DB_VERSION_27
+        ).run {
+            insertDataToTest(this)
         }
 
         helper.runMigrationsAndValidate(
             TEST_DB_NAME,
             DB_VERSION_28,
             true,
             OwncloudDatabase.MIGRATION_27_28
-        )
-
-        validateMigrationTo28()
+        ).also { validateMigrationTo28(it) }
     }
 
     @Test
     fun startInVersion28_containsCorrectData() {
-        with(
-            helper.createDatabase(
-                TEST_DB_NAME,
-                DB_VERSION_28
-            )
-        ) {
+        helper.createDatabase(
+            TEST_DB_NAME,
+            DB_VERSION_28
+        )
+    }
+
+    private fun insertDataToTest(database: SupportSQLiteDatabase) {
+        database.run {
             insert(CAPABILITIES_TABLE_NAME, SQLiteDatabase.CONFLICT_NONE, cv)
             insert(CAPABILITIES_TABLE_NAME, SQLiteDatabase.CONFLICT_NONE, cvWithDefaultValues)
             close()
         }
-
-        validateMigrationTo28()
-    }
-
-    private fun getMigratedRoomDatabase(): OwncloudDatabase {
-        val database = Room.databaseBuilder(
-            ApplicationProvider.getApplicationContext(),
-            OwncloudDatabase::class.java,
-            TEST_DB_NAME
-        )
-            .addMigrations(OwncloudDatabase.MIGRATION_27_28)
-            .build()
-
-        helper.closeWhenFinished(database)
-        return database
     }
 
-    private fun validateMigrationTo28() {
-        val dbCapability =
-            getMigratedRoomDatabase().capabilityDao().getCapabilitiesForAccount(OC_CAPABILITY.accountName!!)
-        with(dbCapability) {
-            assertEquals(OC_CAPABILITY.accountName, accountName)
```

Comment:
All these checks are no longer needed?

Reply:
I had some problems using Dao in migration tests, so i kept it simple.

If migration went well, the database's number of rows should be the same as it was before the migration.